src/Storages/MergeTree/MergeTreeData.h

```diff
@@ -528,7 +528,7 @@ class MergeTreeData : public IStorage, public WithMutableContext
 
     /// If the table contains too many active parts, sleep for a while to give them time to merge.
     /// If until is non-null, wake up from the sleep earlier if the event happened.
-    void delayInsertOrThrowIfNeeded(Poco::Event * until = nullptr) const;
+    void delayInsertOrThrowIfNeeded(Poco::Event * until = nullptr, ContextPtr query_context = nullptr) const;
```

Comment:
Do we really need a default value for arguments now? Looks like all calls of this function with two arguments now.
```suggestion
    void delayInsertOrThrowIfNeeded(Poco::Event * until, ContextPtr query_context) const;
```

Reply:
You are right. It can be removed.